### PR TITLE
Fix field metadata for include attributes on dynamic vector tiles

### PR DESCRIPTION
**Pull request by jterry64**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

Please check the type of change your PR introduces:
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
For dynamic vector tiles, we allow users to specify attributes on the table to include in the vector tiles. This code hasn't been updated to using the new metadata structure, causing vector tiles to fail. We haven't noticed until now because this code is cached until a new latest version is set for the dataset, and I just did that to update to a new version of `nasa_viirs_fire_alerts`.

Issue Number: N/A

## What is the new behavior?
Use new metadata structure in the API.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by solomon-negusse — DISMISSED** [review]

File: app/crud/sync_db/tile_cache_assets.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def get_attributes(dataset: str, version: str, asset_type: str) -> List[Dict[str
         #  needs changes to data-api to assure dynamic vector tile caches
         #  also have the implementation parameter in the creation options
         if tile_cache["dataset"] == dataset and tile_cache["version"] == version:
-            return tile_cache["fields"]
+            return tile_cache["metadata"]["fields"]
```
Comment (solomon-negusse) [inline app/crud/sync_db/tile_cache_assets.py]: Looking at the query, I think this should be `tile_cache["fields"]`

Comment (dmannarino) [reply app/crud/sync_db/tile_cache_assets.py]: Me too: https://github.com/wri/gfw-tile-cache/blob/9c0074ce7f1a7ab202e688ebaaeafcb5d9cce920/app/crud/sync_db/tile_cache_assets.py#L26
Are you sure about this change, @jterry64 ?

Comment (jterry64) [reply app/crud/sync_db/tile_cache_assets.py]: Ah ok, so then I'd need to change `assets.fields as fields` to `asset.metadata->fields as fields` and revert the thing above? `fields` is definitely under `metadata` now

Comment (dmannarino) [reply app/crud/sync_db/tile_cache_assets.py]: That sounds correct to me!

**Review by dmannarino — DISMISSED** [review]

File: app/crud/sync_db/tile_cache_assets.py (hunk 1 above)
Comment (dmannarino) [reply app/crud/sync_db/tile_cache_assets.py]: Sorry, I forgot that assets don't have metadata anymore. Right? Also, we don't do dataset->version->asset metadata inheritance anymore. So should you be setting + getting the version or dataset metadata?

**Pull request closed without merging**